### PR TITLE
Fix broken links in `llvm-coverage-instrumentation.md`

### LINKED ISSUE #2026: `llvm-coverage-instrumentation.md`: borked links
The following errors are related to the file:

```
error: Server returned 404 Not Found for https://github.com/rust-lang/rust/blob/master/tests/mir-opt/instrument_coverage.rs
    ┌─ llvm-coverage-instrumentation.md:316:1
    │
316 │ [`tests/mir-opt/instrument_coverage.rs`].
    │ ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Server returned 404 Not Found for https://github.com/rust-lang/rust/blob/master/tests/mir-opt/instrument_coverage.rs

error: Server returned 404 Not Found for https://github.com/rust-lang/rust/tree/master/tests/run-make/coverage-llvmir
    ┌─ llvm-coverage-instrumentation.md:335:14
    │
335 │ Finally, the [`coverage-llvmir`] test compiles a simple Rust program
    │              ^^^^^^^^^^^^^^^^^^^ Server returned 404 Not Found for https://github.com/rust-lang/rust/tree/master/tests/run-make/coverage-llvmir

error: Server returned 404 Not Found for https://doc.rust-lang.org/nightly/nightly-rustc/rustc_mir_transform/coverage/struct.Instrumentor.html
    ┌─ llvm-coverage-instrumentation.md:359:8
    │
359 │ by the [`Instrumentor`][instrumentor]. For each MIR (each non-const, non-inlined
    │        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Server returned 404 Not Found for https://doc.rust-lang.org/nightly/nightly-rustc/rustc_mir_transform/coverage/struct.Instrumentor.html

error: Server returned 404 Not Found for https://doc.rust-lang.org/nightly/nightly-rustc/rustc_mir_transform/coverage/struct.Instrumentor.html#method.inject_counters
    ┌─ llvm-coverage-instrumentation.md:362:1
    │
362 │ [`inject_counters()`][inject-counters].
    │ ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Server returned 404 Not Found for https://doc.rust-lang.org/nightly/nightly-rustc/rustc_mir_transform/coverage/struct.Instrumentor.html#method.inject_counters

error: Server returned 404 Not Found for https://doc.rust-lang.org/nightly/nightly-rustc/rustc_mir_transform/coverage/spans/struct.CoverageSpans.html#method.generate_coverage_spans
    ┌─ llvm-coverage-instrumentation.md:380:4
    │
380 │ 1. [`generate_coverage_spans()`][generate-coverage-spans] computes the minimum set of distinct,
    │    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Server returned 404 Not Found for https://doc.rust-lang.org/nightly/nightly-rustc/rustc_mir_transform/coverage/spans/struct.CoverageSpans.html#method.generate_coverage_spans

error: Server returned 404 Not Found for https://doc.rust-lang.org/nightly/nightly-rustc/rustc_mir_transform/coverage/counters/struct.BcbCounters.html#method.make_bcb_counters
    ┌─ llvm-coverage-instrumentation.md:383:4
    │
383 │ 2. [`make_bcb_counters()`][make-bcb-counters] generates `BcbCounter::Counter`s and
    │    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Server returned 404 Not Found for https://doc.rust-lang.org/nightly/nightly-rustc/rustc_mir_transform/coverage/counters/struct.BcbCounters.html#method.make_bcb_counters

error: Server returned 404 Not Found for https://doc.rust-lang.org/nightly/nightly-rustc/rustc_mir_transform/coverage/spans/struct.CoverageSpans.html
    ┌─ llvm-coverage-instrumentation.md:468:14
    │
468 │ The `struct` [`CoverageSpans`][coverage-spans] builds and refines a final set of
    │              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Server returned 404 Not Found for https://doc.rust-lang.org/nightly/nightly-rustc/rustc_mir_transform/coverage/spans/struct.CoverageSpans.html

error: Server returned 404 Not Found for https://doc.rust-lang.org/nightly/nightly-rustc/rustc_mir_transform/coverage/spans/struct.CoverageSpan.html
    ┌─ llvm-coverage-instrumentation.md:469:1
    │
469 │ [`CoverageSpan`][coverage-span]s, each representing the largest contiguous `Span`
    │ ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Server returned 404 Not Found for https://doc.rust-lang.org/nightly/nightly-rustc/rustc_mir_transform/coverage/spans/struct.CoverageSpan.html

error: Server returned 404 Not Found for https://doc.rust-lang.org/nightly/nightly-rustc/rustc_mir_transform/coverage/spans/struct.CoverageSpans.html#method.generate_coverage_spans
    ┌─ llvm-coverage-instrumentation.md:474:1
    │
474 │ [`CoverageSpans::generate_coverage_spans()`][generate-coverage-spans] constructs
    │ ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Server returned 404 Not Found for https://doc.rust-lang.org/nightly/nightly-rustc/rustc_mir_transform/coverage/spans/struct.CoverageSpans.html#method.generate_coverage_spans

error: Server returned 404 Not Found for https://doc.rust-lang.org/nightly/nightly-rustc/rustc_mir_transform/coverage/spans/struct.CoverageSpans.html#method.to_refined_spans
    ┌─ llvm-coverage-instrumentation.md:479:1
    │
479 │ [`to_refined_spans()`][to-refined-spans], which iterates through the `CoverageSpan`s,
    │ ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Server returned 404 Not Found for https://doc.rust-lang.org/nightly/nightly-rustc/rustc_mir_transform/coverage/spans/struct.CoverageSpans.html#method.to_refined_spans

error: Server returned 404 Not Found for https://doc.rust-lang.org/nightly/nightly-rustc/rustc_mir_transform/coverage/counters/struct.BcbCounters.html#method.make_bcb_counters
    ┌─ llvm-coverage-instrumentation.md:489:1
    │
489 │ [`make_bcb_counters()`][make-bcb-counters] traverses the `CoverageGraph` and adds a
    │ ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Server returned 404 Not Found for https://doc.rust-lang.org/nightly/nightly-rustc/rustc_mir_transform/coverage/counters/struct.BcbCounters.html#method.make_bcb_counters
```

I noticed the most changes were done by @Zalathar. Can you please take a look at them?

### PR DESCRIPTION
**Pull request by Zalathar**
In some of these cases there is no simple fix, because the text has been invalidated by major structural changes to the code, so instead I've rewritten or deleted those sections.

Fixes #2026.

### CODE REVIEW

File: src/llvm-coverage-instrumentation.md [hunk 1]
```diff
@@ -452,30 +425,9 @@ significance.
...
-that can be used to assign coverage `Counter`s or `Expression`s, one-for-one.
-
-[coverage-spans]: https://doc.rust-lang.org/nightly/nightly-rustc/rustc_mir_transform/coverage/spans/struct.CoverageSpans.html
-[coverage-span]: https://doc.rust-lang.org/nightly/nightly-rustc/rustc_mir_transform/coverage/spans/struct.CoverageSpan.html
-[to-refined-spans]: https://doc.rust-lang.org/nightly/nightly-rustc/rustc_mir_transform/coverage/spans/struct.CoverageSpans.html#method.to_refined_spans
-
 ### `make_bcb_counters()`
 
-[`make_bcb_counters()`][make-bcb-counters] traverses the `CoverageGraph` and adds a
+[`make_bcb_counters`][] traverses the `CoverageGraph` and adds a
```
Comment (jieyouxu) [inline src/llvm-coverage-instrumentation.md:430]: Question: invalid link?

**Review by jieyouxu — APPROVED** [review]
LGTM in general, just a broken link I think

File: src/llvm-coverage-instrumentation.md (hunk 1 above)
Comment (Zalathar) [reply src/llvm-coverage-instrumentation.md:430]: This is shorthand for:

```markdown
[`make_bcb_counters`][`make_bcb_counters`]
```

Comment (tshepang) [reply src/llvm-coverage-instrumentation.md:430]: `[make_bcb_counters]` would also work

Comment (Zalathar) [reply src/llvm-coverage-instrumentation.md:430]: The link already works as-is, so I'm not sure what I'm being asked to do here.

Comment (marxin) [inline src/llvm-coverage-instrumentation.md:430]:
```suggestion
[`make_bcb_counters`] traverses the `CoverageGraph` and adds a
```

The following simplification I guess.

**Comment by jieyouxu** [comment]
Thanks!

**Pull request merged**